topydo/lib/PrettyPrinterFilter.py

```diff
@@ -117,3 +154,97 @@ def filter(self, p_todo_str, _):
+        super(PrettyPrinterBasicPriorityFilter, self).__init__()
+        self.p_replacement = p_replacement
+
+    def filter(self, p_todo_str, _):
+        """ Find the priority """
+        matches = re.search('^(?P<id>\| *\w+\| )?(\((?P<pri>[A-Z])\) )?', p_todo_str)
+        # we are looking for the id string and priority at the beginning of the
+        # line, in the form of:
+        #   | 48| (V) ...
+        #   |h2j| (C) ...
+        if matches:
+            if matches.group('id'):
+                if matches.group('pri'):
+                    """ If we have a priority """
+                    return(matches.group('id') + matches.group('pri') + ' ' + p_todo_str[matches.span()[1]:])
+                else:
+                    return(matches.group('id') + self.p_replacement + ' ' + p_todo_str[matches.span()[1]:])
+            else:
+                if matches.group('pri'):
+                    """ If we have a priority """
+                    return(matches.group('pri') + ' ' + p_todo_str[matches.span()[1]:])
+                else:
+                    return(self.p_replacement + ' ' + p_todo_str[matches.span()[1]:])
+        else:
+            """ No match found """
+            return(self.p_replacement + ' ' + p_todo_str)
+
+
+class PrettyPrinterHumanDatesFilter(PrettyPrinterFilter):
+
+    """ Turns dates to human readable versions. """
+
+    def __init__(self):
+        super(PrettyPrinterHumanDatesFilter, self).__init__()
+
+    def filter(self, p_todo_str, _):
+
+        addingdates = False
+
+        """ First, the date added
+            This, per the spec, will be at the front of the line,
+            after the priority
+        """
+        line2 = p_todo_str
+        pattern2 = re.compile('^(?P<line_start>(\| *\w+\| )?(\(?[A-Z ]\)? )?)(?P<is_date>(?P<year>\d{4})-(?P<month>\d{1,2})-(?P<day>\d{1,2}) )')
+        matches2 = pattern2.match(line2)
+        if matches2:
+            linedate = arrow.now()  # set the time to 'now' so the comparisions work better
+            linedate = linedate.replace(year=int(matches2.group('year')))
+            linedate = linedate.replace(month=int(matches2.group('month')))
+            linedate = linedate.replace(day=int(matches2.group('day')))
+            adddelta = linedate.humanize()
+            line3 = matches2.group('line_start') + pattern2.sub('', line2)
+            addingdates = True
+        else:
+            adddelta = ''
+            line3 = line2
+
+        """ Due dates """
+        pattern3 = re.compile('(?P<is_date> ' + config().tag_due() + ':(?P<year>\d{4})-(?P<month>\d{1,2})-(?P<day>\d{1,2}) ?)')
+        matches3 = pattern3.search(line3)
+        if matches3:
+            linedate = arrow.now()  # set the time to 'now' so the comparisions work better
+            linedate = linedate.replace(year=int(matches3.group('year')))
+            linedate = linedate.replace(month=int(matches3.group('month')))
+            linedate = linedate.replace(day=int(matches3.group('day')))
```

Comment:
The construction of `linedate` looks a lot like the construction above. Suggestion is to put this in a separate (inline) function, such as `get_date_from_match(p_match)`.